ygen/enumgen_test.go

```diff
@@ -20,9 +20,99 @@ import (
 
 	"github.com/google/go-cmp/cmp"
 	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/openconfig/gnmi/errdiff"
 	"github.com/openconfig/goyang/pkg/yang"
 )
 
+func TestResolveDedupSet(t *testing.T) {
+	tests := []struct {
+		name               string
+		inDefinedEnums     map[string]bool
+		inDedupSets        map[string]map[string]bool
+		wantUniqueNamesMap map[string]string
+	}{{
+		name: "no duplication",
+		inDefinedEnums: map[string]bool{
+			"Baz": true,
+		},
+		inDedupSets: map[string]map[string]bool{
+			"Foo": map[string]bool{
+				"enum-a": true,
+			},
+			"Bar": map[string]bool{
+				"enum-b": true,
+			},
+		},
+		wantUniqueNamesMap: map[string]string{
+			"enum-a": "Foo",
+			"enum-b": "Bar",
+		},
+	}, {
+		name: "simple duplication",
+		inDefinedEnums: map[string]bool{
+			"Baz": true,
+		},
+		inDedupSets: map[string]map[string]bool{
+			"Foo": map[string]bool{
+				"enum-a": true,
+				"enum-b": true,
+			},
+		},
+		wantUniqueNamesMap: map[string]string{
+			"enum-a": "Foo",
+			"enum-b": "Foo_",
```

Comment:
This `_` is essentially to keep this change a no-op for existing generated code right?

Reply:
Yes, this test's logic is going change.